### ISSUE TITLE
Snake Case Field Option

### ISSUE THREAD

**Issue by koxudaxi** [koxudaxi/datamodel-code-generator#162]
**Is your feature request related to a problem? Please describe.**
OpenAPI or JsonSchema may describe `key` of objects as `camelCase`.
But, attributes of python should be snake_case.
The generator should provide a feature that converts filed names `CamelCase` to `sanke_case`.

**Describe the solution you'd like**
I would the feature give the generator a command-line option

**Comment by FlorianLudwig** [comment]
I implemented this here:

https://github.com/FlorianLudwig/pdk8s/blob/master/pdk8s/gen_k8s.py#L59

By change the name to camel_case and keeping the old name as alias.

**Comment by koxudaxi** [comment]
Thank you for reviewing the PR.

I have released the version as `0.5.21` :tada:

**Comment by camerondavison** [comment]
Maybe I am off base here but I feel like this is doing the opposite of what I am exepcting.

```python
        class Model(BaseModel):
            notification_type: Optional[str] = Field(
                None,
                alias="notificationType",
            )
        self.assertEqual("{\"notification_type\": \"test\"}", Model(notificationType='test').json())
```

where I was expecting to have
```
self.assertEqual("{\"notificationType\": \"test\"}", Model(notification_type='test').json())
```

Since javascript and json is usually camel case, and python is usually snake case.

**Comment by camerondavison** [comment]
It looks like adding
```python
    class Config:
        allow_population_by_field_name = True
```
and the whenever generating the json doing `.dict(by_alias=True)` works more like I was hoping. Is there a way to add `pydantic` config options to the generated classes?

**Comment by koxudaxi** [comment]
@camerondavison 
I added an option for allow_population_by_field_name
It's included in version 0.5.33
